src/main/java/org/junit/rules/RuleChain.java

```diff
@@ -77,8 +77,12 @@ private RuleChain(List<TestRule> rules) {
      *
      * @param enclosedRule the rule to enclose.
      * @return a new {@code RuleChain}.
+     * @throws IllegalArgumentException if {@code enclosedRule} is null
      */
     public RuleChain around(TestRule enclosedRule) {
+        if (enclosedRule == null) {
+            throw new IllegalArgumentException("The enclosed rule should not be null");
```

Comment:
I see. Done, although I really prefer IllegalArgumentEx as I find it more meaningful than a NPE: NPEs can be thrown by the JVM, whereas IllegalArgEx cannot.
For example, take the unit test: since it's now considering a NPE as expected, I had to add an assertion on the message to make sure that the NPE is really what we are expecting, and not some other null reference (a programming error).